tests/infra_test.go

```diff
@@ -693,12 +696,21 @@ var _ = Describe("[Serial]Infrastructure", func() {
 			}
 
 			errorCount := 0
+			unknownErrorTypeFound := false
 			for ix := 0; ix < concurrency; ix++ {
 				err := <-errors
 				if err != nil {
-					errorCount += 1
+					var e *neturl.Error
+					if errs.As(err, &e) {
+						Expect(e.Err.Error()).To(ContainSubstring("Client.Timeout exceeded while awaiting headers"))
+						errorCount += 1
+					} else {
+						unknownErrorTypeFound = true
```

Comment:
Please try to avoid this flag, it is not serving us well. We loose the error info and it does not read nicely.
I suggest you extract the whole loop to a function and return the error, then assert it.